### PR TITLE
Email component

### PR DESCRIPTION
**Pull request by maxgds**
Adds email component to support topic pages, as per: https://trello.com/c/oRm6qnnb/114-email-alert

screen shot 2018-03-14 at 16 26 59

https://govuk-collections-pr-550.herokuapp.com/component-guide

### CODE REVIEW

File: test/components/email_link_test.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+require 'component_test_helper'
+
+class EmailLinkTest < ComponentTestCase
+  def component_name
+    "email-link"
+  end
+
+  test "renders nothing without a href" do
```
Comment (vanitabarrett) [inline test/components/email_link_test.rb]: We need to add a test that nothing is rendered when nothing is passed to the component too

File: app/views/components/_email-link.html.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+<%
+  text ||= "Sign up for updates to this topic page"
```
[the remark this answers is not in the document]
Comment (tijmenb) [reply app/views/components/_email-link.html.erb]: Could we skip the default and make a `text` required?

**Comment by maxgds** [comment]
@vanitabarrett Have adjusted and removed the default as recommended by @tijmenb - could you take another look?

**Comment by vanitabarrett** [comment]
@maxgds Changes look good 👍 Just spotted this:

Wondering if we need to adjust this to behave like our other icon + text links on GOV.UK, e.g: download link, on smaller screens sizes:

screen shot 2018-03-15 at 09 38 10

vs

screen shot 2018-03-15 at 09 39 27

**Comment by maxgds** [comment]
Good point on the mobile wrapping. I'd checked it looked ok on a few different screens through browserstack, but hadn't realised we had a pattern to make it look better. I'll look at fixing that too, now.

**Comment by vanitabarrett** [comment]
Looks like our download link doesn't actually behave like that - so we can keep this as is.

Looks good 👍

**Pull request merged**